File: simple-message/src/main/java/ru/starksoft/simplemessage/Message.java
package ru.starksoft.simplemessage;

import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.annotation.UiThread;

public interface Message {

	@NonNull
	SimpleMessageManager.Callback getCallback();

	@NonNull
	MessageRecord getMessageRecord();

	void setOnDismissListener(@Nullable OnDismissListener onDismissListener);

	@UiThread
	void show();

	@UiThread
	void hide();

	void hideSystemUi();

	void showSystemUi();

	boolean isDismissing();

	interface OnDismissListener {

		void onDismissed();
	}
}

File: simple-message/src/main/java/ru/starksoft/simplemessage/Message.kt
package ru.starksoft.simplemessage

import android.support.annotation.UiThread

internal interface Message {

	fun getCallback(): SimpleMessageManager.Callback

	fun getMessageData(): MessageData

	fun isDismissing(): Boolean

	fun setOnDismissListener(onDismissListener: OnDismissListener?)

	@UiThread
	fun show()

	@UiThread
	fun hide()

	fun hideSystemUi()

	fun showSystemUi()

	fun showMessageDelay(): Long

	interface OnDismissListener {

		fun onDismissed()
	}
}